docs/production_deployment.md

```diff
@@ -0,0 +1,118 @@
+# Production deployment
+
+## System requirements
+
+Webpack currently requires about 3GB RAM free to build the JavaScript assets in
+production mode, so a server with at least 4GB RAM is likely
+required. Optimisations may be possible, but due to the bootstrap-loader
+integration we can't currently switch to a newer Webpack.
+
+## Production mode
+
+Running `bin/rails server` will by default start Houdini/Rails in "development"
+mode (`Rails.env == "development"`). This flag is used to conditionally
+configure various aspects of the application such as the database settings
+(`config/database.yml`), the in-process Webpack compilation
+(`config/webpacker.yml`), any additional Gems required (`Gemfile`) and others
+(`config/environments/development.rb`, `production.rb` etc.)
+
+To start Houdini in production mode, you can run `bin/rails server
+--environment=production` or run:
+
+```sh
+export RAILS_ENV=production
+bin/rails server
+```
+
+## Configuring the database
+
+See ["Configuring a
+database"](https://guides.rubyonrails.org/configuring.html#configuring-a-database)
+in the Ruby on Rails guide.
+
+## Building assets
+
+In development mode, Houdini's images, CSS and JS are compiled automatically in
+the web server process, but in production mode this should be done beforehand
+(`config.assets.compile = false`). To compile the assets, run:
+
+```sh
+RAILS_ENV=production bin/rails assets:clobber
+RAILS_ENV=production bin/rails assets:precompile
+```
+
+As mentioned above, this step is memory intensive and will likely fail on
+systems with less than 3GB RAM free.
+
+These assets are also not served in production mode by default
```

Comment:
It would be good to also mention that you can use a CDN for the assets, which is a common recommendation: https://guides.rubyonrails.org/asset_pipeline.html#set-up-a-cdn-to-serve-static-assets

Reply:
@wwahammy I've now added a note about CDN setup.

(To be honest, it's not an approach I use myself on web applications. For small-medium traffic sites I find that serving files from Nginx is low-overhead and operationally simple. For high traffic sites, I find that it's useful to put the whole site behind a CDN to provide both caching and some basic security for the origin server. Having the whole site behind the CDN also make it a little easier to troubleshoot since all the traffic is taking the same path.)